File: 263-UglyNumber/263-UglyNumber.js
// Last updated: 3/9/2026, 5:34:23 PM
1/**
2 * @param {number} n
3 * @return {boolean}
4 */
5var isUgly = function (n) {
6    let arr = [2, 3, 5];
7    if (n <= 0) {
8        return false;
9    }
10    for (let value of arr) {
11        while (n % value === 0) {
12            n = n / value;
13        }
14    }
15    return n == 1
16};

File: 263-UglyNumber/263-UglyNumber.cpp
// Last updated: 5/20/2026, 10:01:10 AM
1class Solution {
2public:
3    bool isUgly(int n) {
4        if(n<=0){
5            return false;
6        }
7
8        int arr[3] = {2,3,5};
9
10        for(int i=0;i<3;i++){
11            while(n%arr[i]==0){
12                n=n/arr[i];
13            }
14        } 
15
16        return n==1;
17    }
18};